app/src/main/java/com/spendster/presentation/addExpenses/AddExpensesActivity.java

```diff
@@ -42,15 +53,44 @@ public void onClick(View view) {
         }
     }
 
-    private void launchCalendar(){
-        Toast.makeText(this,"Calendar", Toast.LENGTH_SHORT).show();
+    private void launchCalendar() {
+        Toast.makeText(this, "Calendar", Toast.LENGTH_SHORT).show();
+    }
+
+    private void launchCategoryScreen() {
+        try {
+            startActivityForResult(new Intent(this, ChooseCategoryActivity.class),
+                    REQUEST_CODE_CATEGORY);
+        } catch (NullPointerException np) {
+            Toast.makeText(this, np.getMessage(), Toast.LENGTH_SHORT).show();
+        }
     }
 
-    private void launchCategoryScreen(){
-        Toast.makeText(this,"Category", Toast.LENGTH_SHORT).show();
+    private void launchCurrencyScreen() {
+        Toast.makeText(this, "Currency", Toast.LENGTH_SHORT).show();
     }
 
-    private void launchCurrencyScreen(){
-        Toast.makeText(this,"Currency", Toast.LENGTH_SHORT).show();
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (data != null) {
+            if (resultCode == RESULT_OK) {
+                switch (requestCode) {
+                    case REQUEST_CODE_CALENDAR:
+                        break;
+                    case REQUEST_CODE_CATEGORY:
+                        String json = data.getStringExtra("Category");
+                        Gson gson = new Gson();
+                        Category category = gson.fromJson(json, Category.class);
+                        tvTitle.setText(category.getNameOfCategory());
```

Comment:
@SimonMaha pretty strange that only screen title is changed and category button layout is still "category"